### PR TITLE
normalize common components

### PR DESCRIPTION
**Pull request by zarsky-broad**
With new people coming to the codebase, this is as good a time as any.

My naming rules:
- verb: normal function call (`makeIconButton`)
- noun starting with capital: component, call with `h` (`ButtonPrimary`)
- noun starting lowercase: not a function, reference in place (`spinnerOverlay`)

### CODE REVIEW

File: src/components/common.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ export const buttonPrimary = ({ disabled, ...props }, children) => {
   }, props), children)
```
Comment (panentheos) [inline src/components/common.js]: Hyperscript can't handle single react elements as children, so certain things aren't working (e.g. the terminal link). In many of these components we can now just let `children` pass through with the rest of the props. When that's not possible, we'll need to pass it as an array, e.g. `[children]`

Comment (zarsky-broad) [reply src/components/common.js]: Ah yes, I didn't test this morning because dev was unhappy, so I was expecting _something_ to be weird.

Comment (zarsky-broad) [reply src/components/common.js]: Oh, wait, I should just let them pass implicitly, yes.

**Review by panentheos — APPROVED** [review]
Looks good, thanks!

**Pull request merged**